src/dags/importscripts/import_sport.py

```diff
@@ -0,0 +1,163 @@
+        blake2s("|".join(composite_values).encode(), digest_size=digest_size).hexdigest(), base=16
+    )
+
+
+def read_data(file: str, composite_key: Tuple) -> Generator:
+    """Read the data from csv file and contain it into a data class
+
+    Args:
+        file: the path/to/file.ext that will be read
+        composite_key: Tuple of column name(s) that unqiuely identifiy a record
+
+    Returns:
+        a generator containing the rows of the file plus it's header and the defined composite key
+
+    """
+    dialect = get_dialect(file)
+    header = get_header(file)
+    # setup the generic TypeDict data mold
+    DataObject = TypedDict("DataObject", {col: str for col in header})
+    with open(file, "r") as src:
+        data = csv.reader(src, dialect=dialect)
+        for line in data:
+            # construct lines into dictionary format, to proces into the generic data mold
+            row_dict = {i[0]: i[1] for i in zip(header, line)}
+            row_object = dict(object_=DataObject(row_dict), composite_key=composite_key)
+            yield row_object
+
+
+def generate_unique_id(file: str, composite_key: Tuple, digest_size: int = None) -> Generator:
+    """Generate unique hased id from composite values
+
+    Args:
+        file: the path/to/file.ext that will be read
+        composite_key: Tuple of column name(s) that unqiuely identifiy a record
+        digest_size: size to set the max bytes to use for the hash
+
+    Returns:
+        a generator containing the rows plus the unique ID value for the added ID column
+
+    """
+    for row in read_data(file, composite_key):
+        composite_values = []
+        for column in row["composite_key"]:
+            composite_values.append(row["object_"].get(column, None))
+        try:
+            row["object_"]["id"] = make_hash(composite_values, digest_size=digest_size)
+            yield row["object_"]
+        except TypeError as err:
+            logger.error("%s cannot be retrieved from %s: %s", column, file, err)
+            continue
+
+
+def add_unique_id_to_csv(file: str, composite_key: Tuple, digest_size: int = None) -> None:
+    """save enriched data to csv for further processing in DAG
+
+    Args:
+        file: the path/to/file.ext that will be read
+        composite_key: Tuple of column name(s) that unqiuely identifiy a record
+        digest_size: size to set the max bytes to use for the hash
+
+    Does:
+        saves data rows incl the added new ID column back to the *csv* file
+
+    """
+    data = generate_unique_id(file, composite_key=composite_key)
+    header = next(data)  # get the modified header including the new id column
+    rows = list()
+    for row in data:
+        rows.append(row.values())
+    with open(file, "w") as f:
+        write = csv.writer(f, dialect=csv.unix_dialect)
+        write.writerow(header)
+        write.writerows(rows)
+
+
+def add_unique_id_to_geojson(file: str, composite_key: Tuple, digest_size: int = None) -> None:
```

Comment:
`composite_key: Tuple[str]`

You can remove `digest_size` as you are not using it.

Reply:
Oeps, foutje